Sources/Extensions/SwiftStdlib/IntExtensions.swift

```diff
@@ -171,7 +171,21 @@ public extension Int {
 		return n == 0 ? self : Int(round(Double(self) / Double(n))) * n
 	}
 	// swiftlint:enable identifier_name
-
+    
+    /// SwifterSwift: Array of factors for given integer.
+    /// - Returns: array of factors
+    func factors() -> [Int] {
+        guard self > 0 else { return [] }
+        
+        let rootSquared = Int(Double(self).squareRoot())
+        var factors = Set<Int>()
+        
+        for i in 1...(rootSquared + 1) where self % i == 0 {
```

Comment:
I wouldn't use an enum because it's not friendly in terms of SwifterSwift's "extensions should be easy to copy and paste" philosophy, however, I completely agree that we need to support negative ranges as well. Good catch 👍

Reply:
@freak4pc , do you really think we need negative factors? As for me, it looks like unnecessary overcomplicated. Negative factors equal to positive factors negated. I think providing negative factors leads to less readability. I mean It's implied if we have positive factors, we have the same negative factors

Reply:
I agree with @n0an, there's no need for the negative factors.
This function could be adapted to handle a negative input, I suppose. Thoughts?

Reply:
In that case I’d at least return a UInt. It’s all a question of what you deem as “do we need”? Wha is the use case for this extension? If it is to retrieve _all_ factors then it doesn’t do that correctly ATM 

If there is a “central” use case where these aren’t needed than I’d understand but this is a purely mathematical addition so I’m not sure it’s entirely correct that way

Reply:
Returning an unsigned int to show that we're using non-negative numbers [isn't recommended](https://google.github.io/styleguide/cppguide.html#Integer_Types). Also, this is an extension on `Int`, so returning an array of `UInt` may be confusing to a user of this function.

If you look up any example of finding factors of a number it will always show you the positive results only (e.g. WolframAlpha). As mentioned, the negative results are redundant. I wouldn't consider it to be an incomplete answer if we only have the positive results. Maybe the documentation for the class could be updated to mention this fact.

Reply:
> Also, this is an extension on Int, so returning an array of UInt may be confusing to a user of this function.

Yeah, I wouldn't return a `UInt` here. We can return only positive factors but we should support negative input.